dapp/src/components/AvailabilityCheck.tsx

```diff
@@ -30,14 +33,20 @@ export function AvailabilityCheck() {
         setName(searchValue);
     };
 
-    const handlePurchase = () => {
-        console.log('Purchase initiated for:', searchValue);
-    };
-
     const enableSearch = isValid;
 
     return (
         <div className="flex flex-col items-center w-full space-y-4">
+            {showPurchaseModal && searchValue && (
+                <PurchaseName
+                    name={searchValue}
+                    onClose={() => {
+                        setShowPurchaseModal(false);
+                        setSearchValue('');
+                        setName('');
```

Comment:
better move this to a different function and called it here:
```
    function onClosePurchaseDialog() {
        setShowPurchaseModal(false);
        setSearchValue('');
        setName('');
    }
```
```suggestion
                    onClose={onClosePurchaseDialog}
```